From fa01a83436ad7d80f242690172f96d8d2c4f3c0e Mon Sep 17 00:00:00 2001
From: ValdikSS <iam@valdikss.org.ru>
Date: Mon, 17 Sep 2018 19:04:27 +0300
Subject: [PATCH 2/2] Assume optional codecs are supported if were supported
 previously

This fix is requires to properly follow codec reconfiguration for SBC HD
(SBC Dual Channel) in case of incoming Bluetooth connection.

Change-Id: Idb8fce75b4f628ae865fc1d3a787e34b6a29e31e
---
 src/com/android/bluetooth/a2dp/A2dpService.java | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/com/android/bluetooth/a2dp/A2dpService.java b/src/com/android/bluetooth/a2dp/A2dpService.java
index 0c33f6ac..361ac673 100644
--- a/src/com/android/bluetooth/a2dp/A2dpService.java
+++ b/src/com/android/bluetooth/a2dp/A2dpService.java
@@ -950,11 +950,11 @@ public class A2dpService extends ProfileService {
             }
         }
         if (previousSupport == BluetoothA2dp.OPTIONAL_CODECS_SUPPORT_UNKNOWN
-                || supportsOptional != (previousSupport
-                                    == BluetoothA2dp.OPTIONAL_CODECS_SUPPORTED)) {
+                || previousSupport == BluetoothA2dp.OPTIONAL_CODECS_NOT_SUPPORTED) {
             setSupportsOptionalCodecs(device, supportsOptional);
         }
-        if (supportsOptional) {
+        if (supportsOptional
+                || previousSupport == BluetoothA2dp.OPTIONAL_CODECS_SUPPORTED) {
             int enabled = getOptionalCodecsEnabled(device);
             if (enabled == BluetoothA2dp.OPTIONAL_CODECS_PREF_ENABLED) {
                 enableOptionalCodecs(device);
-- 
2.17.1

